### PR TITLE
set column type

### PR DESCRIPTION
**Pull request by tksfjt1024**
set defauld type for columns

### CODE REVIEW

**Comment by kekekenta** [comment]
preview結果だけ見せていただけると

**Comment by tksfjt1024** [comment]
use_micro_circle = falseの場合
+-------------------+---------------------+-------------+------------------+-------------+------------+---------------+-----------------+-------------------------------+--------------------+-----------------------+---------------------------+
| CampaignId:string | CampaignName:string | Date:string | Impressions:long | Clicks:long | Ctr:double |     Cost:long | AverageCpc:long | ContentImpressionShare:double | Conversions:double | ConversionRate:double | CostPerAllConversion:long |
+-------------------+---------------------+-------------+------------------+-------------+------------+---------------+-----------------+-------------------------------+--------------------+-----------------------+---------------------------+
|        1650523826 |          LP集客 - リタゲ |  2019-12-22 |                0 |           0 |        0.0 |             0 |               0 |                         0.244 |                0.0 |                   0.0 |                         0 |
|        1650523826 |          LP集客 - リタゲ |  2019-12-23 |              908 |           1 |      0.001 |    43,337,893 |      43,337,893 |                         0.151 |                0.0 |                   0.0 |                         0 |
|        1650523826 |          LP集客 - リタゲ |  2019-12-24 |              890 |           1 |      0.001 |    77,945,687 |      77,945,687 |                          0.16 |                0.0 |                   0.0 |                         0 |
|        2057191645 |           20_Embulk |  2019-12-23 |               99 |           2 |       0.02 |   801,000,000 |     400,500,000 |                           0.0 |                0.0 |                   0.0 |                         0 |
|        2057191645 |           20_Embulk |  2019-12-24 |              106 |           2 |      0.019 |   773,000,000 |     386,500,000 |                           0.0 |                0.0 |                   0.0 |                         0 |
|        1650523826 |          LP集客 - リタゲ |  2019-12-25 |              976 |           5 |      0.005 |   308,769,946 |      61,753,989 |                           0.0 |                0.0 |                   0.0 |                         0 |
|        2057196916 |              30_ETL |  2019-12-23 |              209 |           3 |      0.014 | 1,538,000,000 |     512,666,667 |                           0.0 |                0.0 |                   0.0 |                         0 |
|        2057178433 |             10_ブランド |  2019-12-23 |              540 |           7 |      0.013 |   772,000,000 |     110,285,714 |                           0.0 |                0.0 |                   0.0 |                         0 |
|        2057178433 |             10_ブランド |  2019-12-22 |              594 |           5 |      0.008 | 4,256,000,000 |     851,200,000 |                           0.0 |                0.0 |                   0.0 |                         0 |
|        2057196916 |              30_ETL |  2019-12-22 |               48 |           2 |      0.042 | 1,478,000,000 |     739,000,000 |                           0.0 |                0.0 |                   0.0 |                         0 |
|        2057196916 |              30_ETL |  2019-12-24 |              155 |           2 |      0.013 | 1,205,000,000 |     602,500,000 |                           0.0 |                1.0 |                   0.5 |             1,205,000,000 |
|        2057191645 |           20_Embulk |  2019-12-25 |              106 |           2 |      0.019 |   936,000,000 |     468,000,000 |                           0.0 |                0.0 |                   0.0 |                         0 |
|        2057178433 |             10_ブランド |  2019-12-24 |              429 |           8 |      0.019 | 1,137,000,000 |     142,125,000 |                           0.0 |                0.0 |                   0.0 |                         0 |
|        2057196916 |              30_ETL |  2019-12-25 |              139 |           5 |      0.036 | 1,609,000,000 |     321,800,000 |                           0.0 |                0.0 |                   0.0 |                         0 |
|        2057178433 |             10_ブランド |  2019-12-25 |              415 |          11 |      0.027 | 2,054,000,000 |     186,727,273 |                           0.0 |                0.0 |                   0.0 |                         0 |
|        2057191645 |           20_Embulk |  2019-12-22 |               13 |           0 |        0.0 |             0 |               0 |                           0.0 |                0.0 |                   0.0 |                         0 |
+-------------------+---------------------+-------------+------------------+-------------+------------+---------------+-----------------+-------------------------------+--------------------+-----------------------+---------------------------+

**Comment by tksfjt1024** [comment]
use_micro_circle = trueの場合
+-------------------+---------------------+-------------+------------------+-------------+------------+-----------+-----------------+-------------------------------+--------------------+-----------------------+---------------------------+
| CampaignId:string | CampaignName:string | Date:string | Impressions:long | Clicks:long | Ctr:double | Cost:long | AverageCpc:long | ContentImpressionShare:double | Conversions:double | ConversionRate:double | CostPerAllConversion:long |
+-------------------+---------------------+-------------+------------------+-------------+------------+-----------+-----------------+-------------------------------+--------------------+-----------------------+---------------------------+
|        1650523826 |          LP集客 - リタゲ |  2019-12-22 |                0 |           0 |        0.0 |         0 |               0 |                         0.244 |                0.0 |                   0.0 |                         0 |
|        1650523826 |          LP集客 - リタゲ |  2019-12-23 |              908 |           1 |      0.001 |        43 |              43 |                         0.151 |                0.0 |                   0.0 |                         0 |
|        1650523826 |          LP集客 - リタゲ |  2019-12-24 |              890 |           1 |      0.001 |        77 |              77 |                          0.16 |                0.0 |                   0.0 |                         0 |
|        2057191645 |           20_Embulk |  2019-12-23 |               99 |           2 |       0.02 |       801 |             400 |                           0.0 |                0.0 |                   0.0 |                         0 |
|        2057191645 |           20_Embulk |  2019-12-24 |              106 |           2 |      0.019 |       773 |             386 |                           0.0 |                0.0 |                   0.0 |                         0 |
|        1650523826 |          LP集客 - リタゲ |  2019-12-25 |              976 |           5 |      0.005 |       308 |              61 |                           0.0 |                0.0 |                   0.0 |                         0 |
|        2057196916 |              30_ETL |  2019-12-23 |              209 |           3 |      0.014 |     1,538 |             512 |                           0.0 |                0.0 |                   0.0 |                         0 |
|        2057178433 |             10_ブランド |  2019-12-23 |              540 |           7 |      0.013 |       772 |             110 |                           0.0 |                0.0 |                   0.0 |                         0 |
|        2057178433 |             10_ブランド |  2019-12-22 |              594 |           5 |      0.008 |     4,256 |             851 |                           0.0 |                0.0 |                   0.0 |                         0 |
|        2057196916 |              30_ETL |  2019-12-22 |               48 |           2 |      0.042 |     1,478 |             739 |                           0.0 |                0.0 |                   0.0 |                         0 |
|        2057196916 |              30_ETL |  2019-12-24 |              155 |           2 |      0.013 |     1,205 |             602 |                           0.0 |                1.0 |                   0.5 |                     1,205 |
|        2057191645 |           20_Embulk |  2019-12-25 |              106 |           2 |      0.019 |       936 |             468 |                           0.0 |                0.0 |                   0.0 |                         0 |
|        2057178433 |             10_ブランド |  2019-12-24 |              429 |           8 |      0.019 |     1,137 |             142 |                           0.0 |                0.0 |                   0.0 |                         0 |
|        2057196916 |              30_ETL |  2019-12-25 |              139 |           5 |      0.036 |     1,609 |             321 |                           0.0 |                0.0 |                   0.0 |                         0 |
|        2057178433 |             10_ブランド |  2019-12-25 |              415 |          11 |      0.027 |     2,054 |             186 |                           0.0 |                0.0 |                   0.0 |                         0 |
|        2057191645 |           20_Embulk |  2019-12-22 |               13 |           0 |        0.0 |         0 |               0 |                           0.0 |                0.0 |                   0.0 |                         0 |
+-------------------+---------------------+-------------+------------------+-------------+------------+-----------+-----------------+-------------------------------+--------------------+-----------------------+---------------------------+

File: lib/embulk/input/google_adwords.rb [hunk 1]
```diff
@@ -115,6 +122,18 @@ def query_report_results(query, &block)
...
         end
       end
+
+      def formated_row(fields, row, use_micro_circle)
+        fields.each_with_index do |field, i|
+          if %w(Ctr ContentImpressionShare ConversionRate).include?(field)
+            row[i].slice!("%")
+            row[i] = (row[i].to_f * 0.01).round(3)
+          elsif %w(Cost AverageCpc CostPerAllConversion).include?(field) && use_micro_circle
+            row[i] = row[i].to_i / 10 ** 6
```
Comment (kekekenta) [inline lib/embulk/input/google_adwords.rb]: ここfloatになるケースがあるので、丸め込まないでいただけると

Comment (tksfjt1024) [reply lib/embulk/input/google_adwords.rb]: わかりました

File: lib/embulk/input/google_adwords.rb [hunk 2]
```diff
@@ -41,14 +41,21 @@ def self.transaction(config, &control)
...
+          "daterange" => config.param("daterange", :hash, default: {}),
+          "use_micro_circle" => config.param("use_micro_circle", :bool, default: true)
         }
 
         raise ConfigError.new("The parameter report_type must not be empty.") if task["report_type"].empty?
         raise ConfigError.new("The parameter fields must not be empty array.") if task["fields"].empty?
 
         columns = task["fields"].map do |col_name|
-          Column.new(nil, col_name, :string)
+          if %w(Impressions Clicks Cost AverageCpc CostPerAllConversion).include?(col_name)
```
Comment (kekekenta) [inline lib/embulk/input/google_adwords.rb]: Impressions Clicks以外はdoubleですかね

Comment (tksfjt1024) [reply lib/embulk/input/google_adwords.rb]: costもdoubleですか？

File: lib/embulk/input/google_adwords.rb [hunk 3]
```diff
@@ -115,6 +122,18 @@ def query_report_results(query, &block)
           block.call row
         end
       end
+
+      def formated_row(fields, row, use_micro_circle)
+        fields.each_with_index do |field, i|
+          if %w(Ctr ContentImpressionShare ConversionRate).include?(field)
```
Comment (kekekenta) [inline lib/embulk/input/google_adwords.rb]: ContentImpressionShareは `< 10%` というような文字列で返ってくるのでそのまま文字列で大丈夫です。

Comment (tksfjt1024) [reply lib/embulk/input/google_adwords.rb]: わかりました

File: lib/embulk/input/google_adwords.rb (hunk 2 above)
Comment (kekekenta) [reply lib/embulk/input/google_adwords.rb]: https://github.com/trocco-io/embulk-input-google-adwords/pull/2#discussion_r361387089
doubleですね

**Review by kekekenta — APPROVED** [review]

**Comment by kekekenta** [comment]
既存の設定で型変更されないように、フラグ追加をお願いします

**Comment by tksfjt1024** [comment]
@kekekenta 
フラグ追加しました

**Review by kekekenta — APPROVED** [review]

**Pull request merged**